### PR TITLE
Change 'no_iter=' to 'no_iter_no_change='

### PR DESCRIPTION
**Pull request by teamiceberg**
Hello author

This change is needed to amend function params for built-in SGDClassifier module

ppn = SGDClassifier(loss='perceptron', n_iter_no_change=1000)
lr = SGDClassifier(loss='log', n_iter_no_change=1000)
svm = SGDClassifier(loss='hinge', n_iter_no_change=1000)

'no_iter=' param produces error.

### CODE REVIEW

**Comment by rasbt** [comment]
Regarding the suggested change from

    Image(filename='./images/01_01.png', width=500)  --> Image(filename='./code/ch01/images/01_01.png', width=500) 

when I try this on my computers (tried it with the most recent Jupyter Lab on both macOS and Ubuntu), it's not finding the images. I think this is because it will always look for the link relative to the notebook location, not where you run the notebook -- I believe this was different in older versions of jupyter notebook).


Regarding the `n_iter` change, I think this would be more related to the `max_iter` parameter rather than `n_iter_no_change` that was introduced in 0.20. I am a bit hesitant to update it here because it will break earlier versions of scikit-learn then. I think in this case it would be best just to add a note to the code or handle it via version checks like 

```python
if LooseVersion(sklearn_version) >= LooseVersion('0.19'):
    ...
    ppn = SGDClassifier(max_iter=...)
else:
    ppn = SGDClassifier(n_iter=...)
